File: lib/src/reconstruction/reconstruct_replay.dart
import '../replay/replay_runtime.dart';
import 'reconstruct_browser.dart';
import 'reconstruct_graph.dart';
import 'reconstruct_memory.dart';

Map<String, dynamic> reconstructReplayState(Map<String, dynamic> extraction) {
  final replayed = replayRuntimeState(extraction);
  final graph = reconstructRuntimeGraph(extraction);
  final memory = reconstructMemoryFromEnvelope(extraction);
  final browser = reconstructBrowserState(extraction);
  final validation = validateFullRuntimeReplay(extraction, replayed);
  return {
    'replayed': replayed,
    'graph': graph.toJson(),
    'memory': memory,
    'browser': browser,
    'validation': validation,
    'bounded': true,
  };
}

File: src/reconstruction/reconstructReplay.ts
import type { ExtractionEnvelope } from "../contracts/runtimeContracts.js";
import { replayRuntimeState, validateFullRuntimeReplay } from "../replay/replayRuntime.js";
import { reconstructRuntimeGraph } from "./reconstructGraph.js";
import { reconstructMemoryFromEnvelope } from "./reconstructMemory.js";
import { reconstructBrowserState } from "./reconstructBrowser.js";

export function reconstructReplayState(extraction: ExtractionEnvelope): {
  replayed: ExtractionEnvelope;
  graph: ReturnType<typeof reconstructRuntimeGraph>;
  memory: Record<string, unknown>;
  browser: ReturnType<typeof reconstructBrowserState>;
  validation: ReturnType<typeof validateFullRuntimeReplay>;
  bounded: boolean;
} {
  const replayed = replayRuntimeState(extraction);
  const graph = reconstructRuntimeGraph(extraction);
  const memory = reconstructMemoryFromEnvelope(extraction as Record<string, unknown>);
  const browser = reconstructBrowserState(extraction);
  const validation = validateFullRuntimeReplay(extraction, replayed);
  return { replayed, graph, memory, browser, validation, bounded: true };
}
